### PR TITLE
doc: review mesh light samples

### PR DESCRIPTION
**Pull request by greg-fer**
Review and align documentation.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: samples/bluetooth/mesh/light_switch/README.rst [hunk 1]
```diff
@@ -10,20 +10,20 @@ Overview
 
 This sample is split into two source files:
 
-* A main file to handle initialization.
-* One additional file for handling Mesh models
+* A :file:`main.c` file to handle initialization.
+* One additional file for handling Mesh models, :file:`model_handler.c`.
```
Comment (greg-fer) [inline samples/bluetooth/mesh/light_switch/README.rst:15]: @trond-snekvik , can we expand a bit the overview section for this sample?

File: samples/bluetooth/mesh/light/README.rst [hunk 2]
```diff
@@ -5,25 +5,29 @@ Bluetooth: Mesh Light
 
 The Bluetooth Mesh Light sample demonstrates how to set up a basic Mesh server model application and control LEDs with the Bluetooth Mesh, using the :ref:`bt_mesh_onoff_readme`.
 
+This sample is required when working with the :ref:`bluetooth_mesh_light_switch` sample.
```
Comment (greg-fer) [inline samples/bluetooth/mesh/light/README.rst:8]: @trond-snekvik , I'm a bit confused about the relationship with the light switch sample.
If the switch sample is used, does the testing procedure for mesh light changes?

File: samples/bluetooth/mesh/light/README.rst [hunk 3]
```diff
@@ -34,34 +38,39 @@ This sample application has the following composition data:
...
    =================  =================  =================  =================
 
-The :ref:`bt_mesh_onoff_srv_readme` instances in elements 1-4 control LEDs 1-4 respectively.
-The Config server allows configurator devices to configure the node remotely.
-The Health server is used to call attention to the device during provisioning.
+The models are used for the following purposes:
+
+* :ref:`bt_mesh_onoff_srv_readme` instances in elements 1 to 4 each control LEDs 1 to 4, respectively.
+* Config Server allows configurator devices to configure the node remotely.
+* Health Server is used to call attention to the device during provisioning.
```
Comment (greg-fer) [inline samples/bluetooth/mesh/light/README.rst:45]: @trond-snekvik , can you provide more details about the role of Health Server? "call atention" is a bit generic...

File: samples/bluetooth/mesh/light_switch/README.rst [hunk 4]
```diff
@@ -87,40 +99,52 @@ Testing
...
+   #. Leave the rest of the publish parameters at their default values.
+   #. Tap :guilabel:`APPLY` to confirm the configuration.
+
+#. Tap :guilabel:`ON` to light up the first LED on the development kit.
+
+You are now be able to control the first LED on the Mesh Light device by pressing Button 1 on the Mesh Light Switch development kit.
+
+Repeat steps 3-6 for each of the elements on the node to control each of the remaining three LEDs on the Mesh Light device.
 
 Dependencies
```
Comment (greg-fer) [inline samples/bluetooth/mesh/light_switch/README.rst:149]: @trond-snekvik , should the provisioner handler be mentioned in Dependencies?

File: samples/bluetooth/mesh/light_switch/README.rst (hunk 1 above)
Comment (trond-snekvik) [reply samples/bluetooth/mesh/light_switch/README.rst:15]: Yes, good idea. I suggest we make this a separate followup task though.

File: samples/bluetooth/mesh/light/README.rst (hunk 2 above)
Comment (trond-snekvik) [reply samples/bluetooth/mesh/light/README.rst:8]: The test procedure for the light is the same. The last step (interacting with the model) can be skipped, but I don't think it needs to be addressed. Currently the mesh light sample is completely self contained, while the light switch optionally complements it. I think this line is sufficient

File: samples/bluetooth/mesh/light/README.rst (hunk 3 above)
Comment (trond-snekvik) [reply samples/bluetooth/mesh/light/README.rst:45]: Okay. Something like this?

> Health Server provides ``attention`` callbacks that is used to call attention to the device during provisioning, through blinking the device LEDs.

File: samples/bluetooth/mesh/light_switch/README.rst (hunk 4 above)
Comment (trond-snekvik) [reply samples/bluetooth/mesh/light_switch/README.rst:149]: Yes! Good catch.

File: samples/bluetooth/mesh/light_switch/README.rst [hunk 5]
```diff
@@ -10,20 +10,20 @@ Overview
...
-* A main file to handle initialization.
-* One additional file for handling Mesh models
+* A :file:`main.c` file to handle initialization.
+* One additional file for handling Mesh models, :file:`model_handler.c`.
 
 Provisioning
 ============
 
-Provisioning is handled by the :ref:`bt_mesh_dk_prov`.
+Provisioning is handled by the :ref:`bt_mesh_dk_prov`, with the default :file:`prj.conf` configuration.
```
Comment (trond-snekvik) [inline samples/bluetooth/mesh/light_switch/README.rst:19]: Is this addition necessary in this context? I mean, the whole document describes the sample as it behaves in its default configuration. If the users want to do this for their own app, they should instead look at the docs for bt_mesh_dk_prov directly, I think.

File: samples/bluetooth/mesh/light_switch/README.rst [hunk 6]
```diff
@@ -34,21 +34,29 @@ This sample application has the following composition data:
...
 
-The :ref:`bt_mesh_onoff_cli_readme` instances in elements 1-4 are controlled by the buttons on the Device Kit.
-The Config server allows configurator devices to configure the node remotely.
-The Health server is used to call attention to the device during provisioning.
+The models are used for the following purposes:
 
-The model handling is implemented in src/model_handler.c, which uses the :ref:`dk_buttons_and_leds_readme` to detect button presses on the board.
+* :ref:`bt_mesh_onoff_cli_readme` instances in elements 1 to 4 are controlled by the buttons on the development kit.
+* Config Server allows configurator devices to configure the node remotely.
+* Health Server is used to call attention to the device during provisioning.
```
Comment (trond-snekvik) [inline samples/bluetooth/mesh/light_switch/README.rst:41]: Same change here as in the Light readme

File: samples/bluetooth/mesh/light_switch/README.rst [hunk 7]
```diff
@@ -34,21 +34,29 @@ This sample application has the following composition data:
...
+
+The model handling is implemented in :file:`src/model_handler.c`, which uses the :ref:`dk_buttons_and_leds_readme` to detect button presses on the board.
 
 The model handler calls :cpp:func:`bt_mesh_onoff_cli_set` to turn the LEDs of a Mesh Light device on or off.
-As we pass a response buffer to the function, it will will block until the model receives a response from the target device.
-If the function returns successfully, the response buffer will contain the response from the Mesh Light.
-In this example, the contents of the response will be ignored, as we handle all incoming OnOff status messages in the status callback.
-Note that the response structure is still required, as the Generic OnOff Client model won't request a response from the Server if we pass ``NULL`` in the response parameter.
+As a response buffer is passed to the function, it blocks until the model receives a response from the target device.
+If the function returns successfully, the response buffer will contain the response from Mesh Light.
```
Comment (trond-snekvik) [inline samples/bluetooth/mesh/light_switch/README.rst:47]: > ... the response from Mesh Light

the response from the Mesh Light device? It's the device that sends the response, not the example, if you get what I mean

File: samples/bluetooth/mesh/light_switch/README.rst [hunk 8]
```diff
@@ -34,21 +34,29 @@ This sample application has the following composition data:
...
+In this sample, the contents of the response will be ignored, as all incoming OnOff status messages are handled in the status callback.
+
+.. note::
+    The response structure is mandatory, because the Generic OnOff Client model will not request a response from the Server if ``NULL`` is passed in the response parameter.
```
Comment (trond-snekvik) [inline samples/bluetooth/mesh/light_switch/README.rst:51]: This is actually outdated now! I need to fix the sample, so I can remove this note as part of that task.

No action required in this PR, I think.

File: samples/bluetooth/mesh/light_switch/README.rst [hunk 9]
```diff
@@ -87,40 +99,52 @@ Testing
...
+
+#. Configure the Client model publish parameters, which define how the model will send its messages:
+
+   1. Tap :guilabel:`SET PUBLICATION`.
+   #. Set the Publish Address to the first unicast address of the Mesh Light node.
+   #. Set the Retransmit Count to zero (:guilabel:`Disabled`) to prevent the model from sending each button press multiple times.
+   #. Leave the rest of the publish parameters at their default values.
+   #. Tap :guilabel:`APPLY` to confirm the configuration.
+
+#. Tap :guilabel:`ON` to light up the first LED on the development kit.
```
Comment (trond-snekvik) [inline samples/bluetooth/mesh/light_switch/README.rst:143]: This step shouldn't be part of the light switch guide. It does not work for this sample, it's only for the mesh light.

File: samples/bluetooth/mesh/light_switch/README.rst [hunk 10]
```diff
@@ -87,40 +99,52 @@ Testing
...
+
+You are now be able to control the first LED on the Mesh Light device by pressing Button 1 on the Mesh Light Switch development kit.
+
+Repeat steps 3-6 for each of the elements on the node to control each of the remaining three LEDs on the Mesh Light device.
```
Comment (trond-snekvik) [inline samples/bluetooth/mesh/light_switch/README.rst:147]: Remember to change to 3-5 if you remove the last point.

Comment (greg-fer) [reply samples/bluetooth/mesh/light_switch/README.rst:147]: Thanks!

File: samples/bluetooth/mesh/light_switch/README.rst (hunk 5 above)
Comment (greg-fer) [reply samples/bluetooth/mesh/light_switch/README.rst:19]: Discussed f2f and removed here and in light readme.

File: samples/bluetooth/mesh/light/README.rst (hunk 3 above)
Comment (greg-fer) [reply samples/bluetooth/mesh/light/README.rst:45]: Discussed f2f and edited a bit more.
